src/components/views/rooms/AudioRecorder.js

```diff
@@ -0,0 +1,111 @@
+//@flow
+
+import React from 'react';
+import classNames from 'classnames';
+
+import sdk from '../../../index';
+import Modal from '../../../Modal';
+import ErrorDialog from '../dialogs/ErrorDialog';
+
+export default class AudioRecorder extends React.Component {
+  static propTypes = {
+    sendAudioBlob: React.PropTypes.func.isRequired,
+  };
+
+  mediaRecorder: ?MediaRecorder = null;
+  blobs: Array<Blob> = [];
+
+  constructor(props: Object) {
+    super(props);
+
+    this.state = {
+      recording: false,
+    };
+  }
+
+  onRecordClicked = async () => {
+    if(this.state.recording) {
+      this.mediaRecorder.stop();
+      this.mediaRecorder = null;
+      this.mediaStream.getTracks().forEach((track) => track.stop());
+      this.mediaStream = null;
+    } else {
+      if(!navigator.mediaDevices || !MediaRecorder) {
+        Modal.createDialog(ErrorDialog, {
+          description: `Sorry, but the features required for push 
+ to talk messages are not supported by your browser`,
+        });
+        return;
+      }
+
+      try {
+        const mediaStream = await navigator.mediaDevices.getUserMedia({
+          audio: true,
+          video: false,
+        });
+        this.onUserMediaSuccess(mediaStream);
+      } catch (e) {
+        this.onUserMediaFailure(e);
+      }
+    }
+  };
+
+  onUserMediaFailure = () => {
+    Modal.createDialog(ErrorDialog, {
+      description: "Failed to start recording.",
+    });
+  };
+
+  onUserMediaSuccess = (mediaStream: MediaStream) => {
+    this.mediaStream = mediaStream;
+
+    this.mediaRecorder = new MediaRecorder(mediaStream);
+    this.mediaRecorder.addEventListener('dataavailable',
+      this.onMediaRecorderDataAvailable);
+    this.mediaRecorder.addEventListener('stop',
+      this.onMediaRecorderStop);
+    this.mediaRecorder.start(1000);
+
+    this.setState({recording: true});
+  };
+
+  onMediaRecorderDataAvailable = (event: BlobEvent) => {
+    this.blobs.push(event.data);
+  };
+
+  onMediaRecorderStop = () => {
+    // Chrome gives us video/webm even though we've requested audio
+    // for some reason. The data's still just audio though.
```

Comment:
It might be more sensible if given a mimeType in the MediaRecorder constructor

Reply:
Seems like Chrome only supports recording WebM, see https://stackoverflow.com/questions/41739837/all-mime-types-supported-by-mediarecorder-in-firefox-and-chrome and links in the answer. Is that important?